tempto-core/src/main/java/com/teradata/tempto/internal/hadoop/hdfs/HdfsModuleProvider.java

```diff
@@ -36,10 +44,49 @@ public Module getModule(Configuration configuration)
             @Override
             protected void configure()
             {
-                bind(HdfsClient.class).to(WebHDFSClient.class).in(Scopes.SINGLETON);
+                bind(HdfsClient.class).toInstance(createHdfsClient());
```

Comment:
instead binding to instance here cleaner approach is to install module providing SpngoHdfsClient or SimpleHdfsClient based on config value.